src/components/Test.vue

```diff
@@ -105,11 +114,21 @@ export default {
       document.addEventListener('keydown', this._keyListener.bind(this));
     },
 
+    async updatePreference (params) {
+      const newTopics = this.topics;
+      newTopics.push(params)
```

Comment:
the existing topic for that technology has to be replaced...
topicIndex = this.topics.findIndex(t -> t.technology === updatedTopic.technology);
this.topics.splice(topicIndex, 1, updatedTopic);